### PR TITLE
Fix transaction and rollback

### PR DESCRIPTION
**Pull request by fogelito**

### CODE REVIEW

File: src/Database/Adapter/MariaDB.php [hunk 1]
```diff
@@ -824,7 +829,11 @@ public function createDocument(string $collection, Document $document): Document
 
             $this->getPDO()->commit();
         } catch (\Throwable $e) {
-            $this->getPDO()->rollBack();
+            if($this->getPDO()->inTransaction()) {
+                var_dump("---- rollBack before ---- ");
```
Comment (abnegate) [inline src/Database/Adapter/MariaDB.php:833]: Let's remove the var_dumps

File: src/Database/Adapter/MariaDB.php [hunk 2]
```diff
@@ -715,105 +715,110 @@ public function deleteIndex(string $collection, string $id): bool
...
+            if (!empty($document->getInternalId())) {
+                $bindKey = '_id';
+                $columns .= "_id, ";
+                $columnNames .= ':' . $bindKey . ', ';
+            }
+
+            $sql = "
 			INSERT INTO {$this->getSQLTable($name)} ({$columns} _uid)
 			VALUES ({$columnNames} :_uid)
 		";
```
Comment (abnegate) [inline src/Database/Adapter/MariaDB.php:758]: Let's fix the indent here

Comment (fogelito) [reply src/Database/Adapter/MariaDB.php]: Done!

File: src/Database/Adapter/MariaDB.php (hunk 1 above)
Comment (fogelito) [reply src/Database/Adapter/MariaDB.php]: Done!

**Review by abnegate — APPROVED** [review]

**Pull request merged**